### ISSUE TITLE
default version of ts and ng in extension

### ISSUE THREAD

**Issue by kyliau** [angular/vscode-ng-language-service#556]
https://github.com/angular/vscode-ng-language-service/pull/552 re-enabled the feature to load TypeScript (abbrv ts) and @angular/language-service (abbrv ng) from user's workspace.

This is good for consistency, but based on many issues we have seen in the past, it does not provide the best user experience.

Take for example a typical upgrade story:
1. extension is current v1 with ng v1
2. we release extension v1.1 with ng v1.1
3. user downloads and installs new extension in vscode
4. without upgrading to ng v1.1 in their *workspace*, users don't see new features or bug fixes at all

This is because we currently prioritize ts and ng in the user project, and use the bundled version only as a fallback if it is not found in user project. 
In short, upgrade is a two step process which is not apparent to many users, and it does not happen automatically.

Suggestions:
1. Flip the default (prioritize bundled version), but provide a way for users to pick custom version (similar to TS, shown in screenshot below). This would require us to add a UI element to vscode.
2. Ideally, we should use the TS version that user currently picked in the editor, but afaik there's no way to obtain such information (we could read tsdk config value, but not the currently selected ts version).

![vscode](https://user-images.githubusercontent.com/2941178/72465250-af584180-378b-11ea-8d9c-7e60b7394a91.png)

**Comment by ayazhafiz** [comment]
> Ideally, we should use the TS version that user currently picked in the editor, but afaik there's no way to obtain such information (we could read tsdk config value, but not the currently selected ts version).

Maybe it is enough to check the tsdk (if it exists, the user is probably using it). We could also add a UI element for selecting the TypeScript version to be used by the language service extension (VS Code's version or the workspace version, identical to the TSLS features dropdown).

I think we should also publicly document that the extension will pick up language-service versions installed in the workspace. This is mentioned in the wiki, but not the README.

**Comment by kyliau** [comment]
Work items here are:

1. Flip the default to prioritize bundled version over workspace version.
2. Register a new command to let users pick their workspace versions (both ts and ng), similar to TS screenshot above. The new command will be available in the editor UI.
3. Document the feature in README and CHANGELOG.

**Comment by ayazhafiz** [comment]
@kyliau is this done?

**Comment by kyliau** [comment]
@ayazhafiz (2) is not done yet since the only way to provide a custom version of Angular and TypeScript is via the configuration settings, not a vscode command. That said, we have moved away from the model of loading custom versions since that hasn't worked well in the past (lots of compatibility issues since @angular/language-service is designed to work with a particular TS version). I think we should close this for now, and revisit this feature when we do get legit requests from users.

It's also worth noting that this is a feature for power-users, and for them the configuration setting is sufficient.
